### PR TITLE
Add first draft of the qabel register spec

### PR DESCRIPTION
**Pull request by jasinai**
Questions:
* Ok, that e-mail and mobile number are optional?
* Will update work? Sign with private key? => No account/password
* Would you expect other Request/Response data?
* In an earlier version of qabel-register, we had the idea, that you get a url to the identity, so you could print it on your business card. Do we still want this?
* Should an alias be unique?
* Other things I didn't mention or forgot?

Closes qabel/qabel-register#1

### CODE REVIEW

File: _docs/Qabel-Protocol-Register.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+title: "Protocol: register"
+---
+# Register Specification
+
+## Abstract
+
+Protocols to store and search information about the users' identities on a server.
```
Comment (jan-schreib) [inline _docs/Qabel-Protocol-Register.md]: In Components-Register.md its "user's" (singular) here it is "users'" (plural) wouldn't it be better to keep it consistend?

File: _docs/Qabel-Protocol-Register.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Used services
+
+Qabel Register uses a server with a database.
+
+## Register server
+
+The register server receives a new identity and saves it to a database, if the identity is valid. The request must contain an alias, the drop url and the public key, addtionally it can contain an email address and a telephone number. Every user can search for an identity on this server, if the request contains at least one of the previous information.
```
Comment (jan-schreib) [inline _docs/Qabel-Protocol-Register.md]: typo "additionally" instead of "addtionally"

**Comment by audax** [comment]
Contacts can have multiple drop urls, so should we allow them to save multiple urls on the register server?

**Comment by schulze** [comment]
> * Will update work? Sign with private key? => No account/password

The secret/public key we have is only used for DH so far. While it can in principle also be used for signatures (after conversion to the Ed25519 key format), it would not be a good idea to use the same key for multiple purposes.

 I believe that a simple solution would be to provide the register component with a Noise key pair, so that update and delete request can be send as encrypted and authenticated boxes.

Another possibility would be to add a Ed25519 key pair for signatures to the client. This will probably also be useful for other use cases that need signatures for non-deniability. This would need a little bit of additional C and Java code in the client, but there are also already [numerous](http://ianix.com/pub/ed25519-deployment.html) Java and Python implementations and wrapper. We would then also need to link the signature key with public key used for encryption, e.g. by signing the encryption key with the signature key (like in an OpenPGP key). 

(Additional bonus for @jan-schreib: signify also uses Ed25519 for its signatures.)

**Comment by audax** [comment]
The enhancement mentioned by @schulze is in #5.

**Comment by audax** [comment]
Also: ack

**Comment by thechauffeur** [comment]
I like this spec and ack this. However, the current implementation is not compliant to this spec. The two things which needs to be discussed are is it OK or not to `search` for drop urls and how / if should the `update` (and maybe also the delete) API call be implemented.

**Comment by jan-schreib** [comment]
@thechauffeur are there any news about your mentions?

What are the plans for the register server?

**Comment by julianseeger** [comment]
This issue should be deferred to next week, there is a meeting about the register server that should have major influence on the spec draft

**Pull request closed without merging**